### DISCUSSION TITLE
Status of Open Policy Agent integration

### DISCUSSION THREAD

**Discussion by betweenstrap21** [minio/minio#12422] [Q&A] [answered]
Hello,

There was a time when the use of Open Policy Agent with STS was documented (https://github.com/minio/minio/blob/RELEASE.2018-11-30T03-56-59Z/docs/sts/opa.md) but it is not the case anymore. The OPA integration has been moved in a "legacy" folder of the code. What is the reason behind not supporting OPA anymore ? Will the OPA support come back in a future release ? Will the legacy part of the code be removed in future versions ? 
I have the requirement of using OPA because, in my case, it is not possible to have the policy name encoded in the access token provided by my IDP (currently having the policy name encoded in the access token seems to be the only alternative left)...

Thank you

Regards

**Comment by harshavardhana** [comment, marked as answer]
> Hello,
> 
> There was a time when the use of Open Policy Agent with STS was documented (https://github.com/minio/minio/blob/RELEASE.2018-11-30T03-56-59Z/docs/sts/opa.md) but it is not the case anymore. The OPA integration has been moved in a "legacy" folder of the code. What is the reason behind not supporting OPA anymore ? Will the OPA support come back in a future release ? Will the legacy part of the code be removed in future versions ?
> I have the requirement of using OPA because, in my case, it is not possible to have the policy name encoded in the access token provided by my IDP (currently having the policy name encoded in the access token seems to be the only alternative left)...
> 
> Thank you
> 
> Regards

It is kept for backward compatibility, we will not expand the scope of OPA support anymore.

There are no plans to further spend time on it.

**Reply by rikatz** [reply]
@harshavardhana as someone looking into this as well, my question is if there's some plan to support the integration with any admission webhook (the same way as k8s), like sending a full JSON to some webhook that would unmarshall that and only answer with yes/no and a reason?

**Reply by harshavardhana** [reply]
@rikatz no that will not be implemented.
